pkg/plugingetter/getter.go

```diff
@@ -21,6 +24,14 @@ type CompatPlugin interface {
 	IsV1() bool
 }
 
+// PluginAddr is a plugin that exposes the socket address for creating custom clients rather than the built-in `*plugins.Client`
+type PluginAddr interface {
+	CompatPlugin
+	Addr() net.Addr
+	Timeout() time.Duration
```

Comment:
Why is timeout part of the plugin's interface? Timeout is something that the user should be allowed to configure (as it exists now), not part of the plugin specification.